### PR TITLE
[cosmosdb-preview] Add support for new api version 2024-02-15-preview

### PR DESCRIPTION
**Pull request by carjackson-msft**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
cosmosdb-preview

**Comment by yonzhan** [comment]
Please fix CI issues

File: src/cosmosdb-preview/setup.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 # TODO: Confirm this is the right version number you want and it matches your
 # HISTORY.rst entry.
-VERSION = '0.26.0'
+VERSION = '0.27.0'
```
Comment (evelyn-ys) [inline src/cosmosdb-preview/setup.py:19]: There're breaking changes in this PR so you may reconsider the version

Comment (evelyn-ys) [reply src/cosmosdb-preview/setup.py:19]: See https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md

**Comment by evelyn-ys** [comment]
Please fix CI failure on style check

**Comment by evelyn-ys** [comment]
Pls upgrade your local `azdev` to latest version and run `azdev style` again. Or you can check the failure details in CI logs

File: src/cosmosdb-preview/HISTORY.rst [hunk 2]
```diff
@@ -2,12 +2,22 @@
 
 Release History
 ===============
+0.27.0
```
Comment (evelyn-ys) [inline src/cosmosdb-preview/HISTORY.rst:5]: Oh you forgot to change the version in history notes

**Review by evelyn-ys — CHANGES_REQUESTED** [review]
Please update the version in `HISTORY.rst` to align with the version in `setup.py`

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**